alacritty_terminal/src/renderer/mod.rs

```diff
@@ -286,7 +295,76 @@ impl GlyphCache {
         FontDesc::new(desc.family.clone(), style)
     }
 
-    pub fn get<'a, L>(&'a mut self, glyph_key: GlyphKey, loader: &mut L) -> &'a Glyph
+    //pub fn font_metrics(&self) -> font::Metrics {
+    //    self.rasterizer
+    //        .metrics(self.font_key, self.font_size)
+    //        .expect("metrics load since font is loaded at glyph cache creation")
+    //}
+
+    // Since shaping is not available on Windows/Mac OSX, text run glyphs are rasterized one by one
+    // as characters
```

Comment:
https://github.com/yeslogic/allsorts ?

Reply:
We're open to it if the performance works well and it properly supports all necessary features on the platforms it could be used for.

However someone would actually have to do the work to try that out of course.

Reply:
And please, avoid pinging folks without a reason for it. thx.

Reply:
The reason is that Weslay, if I recall correctly, had offered working together with anyone who would want to pick this up.

Reply:
yeah, and if anyone had a code with the mentioned lib and they had problems with it then maybe it might be something where pinging could make sense.

Reply:
Or when trying to summon the wizards before the raid, but that arguably doesn't always work :wink:

Reply:
Hello all, I would be happy to answer any allsorts related queries. There should be no cross-platform issues — there are no OS dependencies except libc. We ship builds of Prince, which uses allsorts on at least Linux, macOS, Windows, and FreeBSD. I will admit the the API is currently quite low level, we hope to eventually add an easier to use higher-level API, but have not had a chance to do that yet.

Reply:
Allsorts is certainly looking like a great solution for supporting all platforms at the same time. So the primary question would be how well it performs and how nicely the API integrates with Alacritty, but I think generally a lower level API should be a pretty good fit for Alacritty.

Since you're already around @wezm, I'd imagine you might be aware of how allsorts compares to harfbuzz in performance? Specifically for running relatively simple combination of characters like ligatures on short-ish strings.

Reply:
>  I'd imagine you might be aware of how allsorts compares to harfbuzz in performance? Specifically for running relatively simple combination of characters like ligatures on short-ish strings.

We have not done a lot of comparison with Harfbuzz at this point as the primary focus was accurate shaping, and matching or exceeding the performance of our previous shaping engine. There is an open issue to explore this further https://github.com/yeslogic/allsorts/issues/2

Whilst quite fast, it's probably safe to assume that Allsorts is not as fast as Harfbuzz, as Harfbuzz has been optimised for performance over many years. That is not to say we ignored performance though. We have profiled and optimised it in order to meet, and then exceed the performance of shaping engine it replaced in Prince.